addons/mail/mail_followers.py

```diff
@@ -55,20 +55,29 @@ class mail_followers(osv.Model):
     # Modifying followers change access rights to individual documents. As the
     # cache may contain accessible/inaccessible data, one has to refresh it.
     #
-    def create(self, cr, uid, vals, context=None):
-        res = super(mail_followers, self).create(cr, uid, vals, context=context)
-        self.invalidate_cache(cr, uid, context=context)
+    @api.multi
+    def _invalidate_documents(self):
+        for record in self:
+            if record.res_id:
+                self.env[record.res_model].invalidate_cache(ids=[record.res_id])
+
+    @api.model
+    def create(self, vals):
+        record = super(mail_followers, self).create(vals)
+        record._invalidate_documents()
+        return record
+
+    @api.multi
+    def write(self, vals):
+        self._invalidate_documents()
```

Comment:
I suppose there is an invalidate before and after in case we update res_model / res_id and impact records may differ ?